service/options.go

```diff
@@ -26,6 +26,14 @@ func WithLogger(log logr.Logger) Option {
 func WithFilters(filters ...filter.Filter) Option {
 	return optionFunc(func(svc *ExtProcessor) {
 		svc.filters = filters
+
+		var streams []filter.Stream
+		for _, f := range filters {
+			if s, ok := f.(filter.Stream); ok {
+				streams = append(streams, s)
+			}
+		}
+		svc.streams = streams
```

Comment:
Wasn't sure about putting this here - I think it makes sense (options configure server, so it matches the responsibility).